appengine/standard/firebase/firetactoe/static/main.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/**
+ * @fileoverview Tic-Tac-Toe, using the Firebase API
+ */
+
+/**
+ * @param gameKey - a unique key for this game.
+ * @param me - my user id.
+ * @param token - secure token passed from the server
+ * @param channelId - id of the 'channel' we'll be listening to
+ */
+function initGame(gameKey, me, token, channelId, initialMessage) {
+  var state = {
+    gameKey: gameKey,
+    me: me
+  };
+
+  // This is our Firebase realtime DB path that we'll listen to for updates
+  // We'll initialize this later in openChannel()
+  var channel = null;
+
+  /**
+   * Updates the displayed game board.
+   */
+  function updateGame() {
+    for (var i = 0; i < 9; i++) {
+      var square = document.getElementById(i);
```

Comment:
(not to say there might not be wins to be gained with a larger refactor... (/me eyes the manual style setting below) but it's non-obvious and the clock is ticking :-/)